ui/packages/app/web/src/components/ProfileExplorer.tsx

```diff
@@ -11,6 +11,9 @@ interface ProfileExplorerProps {
 }
 
 const ProfileExplorer = ({router, queryClient}: ProfileExplorerProps): JSX.Element => {
+  if (!router.isReady) {
```

Comment:
During the first render, the `next/router` is returning empty query params(even though the query params are there in the URL), probably because the app is still bootstrapping. This is causing the component state to go out of sync from the query params. So added this check to prevent that. 

Ref: https://github.com/vercel/next.js/discussions/11484

A couple of options without adding this check:
1. To watch for the router queryParam object changes and sync the component state with it.
2. To getter function that handles both the queryParam object and the component state and pick the right one. This was the previous approach we used [here](https://github.com/parca-dev/parca/pull/581/files#diff-54a3b6e9068552cd801e2def85a82c0c1636c06bee1ed183ae357368742d5bbeL285-L293).

I don't feel so good about both of these options, so added a check at the root level. 

Please let me know your thoughts.

Reply:
that makes sense. was just curious as to why it was added. thanks for the explainer!